### PR TITLE
Added multi-build for net472 (client) and netstandard 2.0 (server)

### PR DESCRIPTION
**Pull request by Measurity**
- [x] ~~Find alternative serializer for .netstandard 2.0, BinaryPack supports 2.1 only and Nitrox.BinaryPack for net472 does not work with 2.0 as is.~~ Issue found by TwinBuilderOne, breaking change in .NET 5+ caused wrong union order for packets due to culture string compare: <https://learn.microsoft.com/en-us/dotnet/standard/base-types/string-comparison-net-5-plus>
- [x] Ensure build process is correct (NitroxLauncher depends on Server.exe, but can not reference it as a project because launcher runs on net472 at the moment.)

### CODE REVIEW

**Comment by Measurity** [comment]
This PR is ready for review

**Review by tornac1234 — CHANGES_REQUESTED** [review]
There are some new warnings happening after this PR modifications:
```cs
6>[...]\NitroxServer-Subnautica\Resources\Parsers\RandomStartParser.cs(31,28,31,58): warning CA1416: This call site is reachable on all platforms. 'RotateFlipType.RotateNoneFlipY' is only supported on: 'windows'.
6>[...]\NitroxServer-Subnautica\Resources\Parsers\RandomStartParser.cs(31,9,31,59): warning CA1416: This call site is reachable on all platforms. 'Image.RotateFlip(RotateFlipType)' is only supported on: 'windows'.
6>[...]\NitroxServer-Subnautica\Resources\Parsers\RandomStartParser.cs(29,98,29,125): warning CA1416: This call site is reachable on all platforms. 'PixelFormat.Format32bppArgb' is only supported on: 'windows'.
6>[...]\NitroxServer-Subnautica\Resources\Parsers\RandomStartParser.cs(29,26,30,80): warning CA1416: This call site is reachable on all platforms. 'Bitmap' is only supported on: 'windows'.
6>[...]\NitroxServer-Subnautica\Resources\AddressablesTools\Catalog\SerializedObjectDecoder.cs(64,28,64,64): warning CA1416: This call site is reachable on all platforms. 'Type.GetTypeFromCLSID(Guid)' is only supported on: 'windows'.
```

**Comment by tornac1234** [comment]
Also breaks tests

**Comment by 20PercentRendered** [comment]
For the servers to work on linux, System.Drawing needs to be replaced with SkiaSharp or ImageSharp. 
If you just want the warnings fixed, you can add IFs around the code that uses System.Drawing. This isn't the most ideal solution though, as it does add a bunch of if statements into the code.

**Comment by Measurity** [comment]
> For the servers to work on linux, System.Drawing ...
This is something I'll look into in the `Linux` branch once this is merged.

File: Nitrox.BuildTool/Nitrox.BuildTool.csproj [hunk 1]
```diff
@@ -17,4 +17,3 @@
     </ItemGroup>
 
 </Project>
-
```
[the remark this answers is not in the document]
Comment (Measurity) [reply Nitrox.BuildTool/Nitrox.BuildTool.csproj:20]: All files should end with an empty new line. Looking in my IDE for this file, it does have one.

File: NitroxLauncher/NitroxLauncher.csproj [hunk 2]
```diff
@@ -9,47 +9,54 @@
...
+            <Visible>false</Visible>
+        </None>
     </ItemGroup>
 
-
     <ItemGroup>
-      <ProjectReference Include="..\NitroxPatcher\NitroxPatcher.csproj" />
-      <ProjectReference Include="..\NitroxServer-Subnautica\NitroxServer-Subnautica.csproj" />
+        <ProjectReference Include="..\NitroxPatcher\NitroxPatcher.csproj" />
+        <ProjectReference Include="..\NitroxServer\NitroxServer.csproj" />
```
[the remark this answers is not in the document]
Comment (Measurity) [reply NitroxLauncher/NitroxLauncher.csproj:40]: The launcher depends on NitroxServer code (which was previously included via NitroxServer-Subnautica reference). But now the launcher builds the server and its a different .net, so I can't reference the server directly without causing warnings (and potentially logic errors in code).

**Comment by Measurity** [comment]
> There are some new warnings happening after this PR modifications:
> 
> ```cs
> 6>[...]\NitroxServer-Subnautica\Resources\Parsers\RandomStartParser.cs(31,28,31,58): warning CA1416: This call site is reachable on all platforms. 'RotateFlipType.RotateNoneFlipY' is only supported on: 'windows'.
> 6>[...]\NitroxServer-Subnautica\Resources\Parsers\RandomStartParser.cs(31,9,31,59): warning CA1416: This call site is reachable on all platforms. 'Image.RotateFlip(RotateFlipType)' is only supported on: 'windows'.
> 6>[...]\NitroxServer-Subnautica\Resources\Parsers\RandomStartParser.cs(29,98,29,125): warning CA1416: This call site is reachable on all platforms. 'PixelFormat.Format32bppArgb' is only supported on: 'windows'.
> 6>[...]\NitroxServer-Subnautica\Resources\Parsers\RandomStartParser.cs(29,26,30,80): warning CA1416: This call site is reachable on all platforms. 'Bitmap' is only supported on: 'windows'.
> 6>[...]\NitroxServer-Subnautica\Resources\AddressablesTools\Catalog\SerializedObjectDecoder.cs(64,28,64,64): warning CA1416: This call site is reachable on all platforms. 'Type.GetTypeFromCLSID(Guid)' is only supported on: 'windows'.
> ```

Those warnings are new because the code can now run on Linux but it still uses some windows only code. It is something I will fix in the Linux branch.

File: Nitrox.Test/Nitrox.Test.csproj [hunk 3]
```diff
@@ -10,10 +10,12 @@
...
         <PackageReference Include="AutoBogus.NSubstitute" Version="2.13.1" />
         <PackageReference Include="CompareNETObjects" Version="4.78.0" />
-        <PackageReference Include="coverlet.collector" Version="3.1.0" />
-        <PackageReference Include="FluentAssertions" Version="6.2.0" />
-        <PackageReference Include="MSTest" Version="3.0.2" />
-        <PackageReference Include="NSubstitute" Version="4.2.2" />
+        <PackageReference Include="FluentAssertions" Version="6.9.0" />
+        <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.3.2" />
+        <PackageReference Include="MSTest.TestAdapter" Version="3.0.2" />
+        <PackageReference Include="MSTest.TestFramework" Version="3.0.2" />
```
Comment (Jannify) [inline Nitrox.Test/Nitrox.Test.csproj:16]: Why are you splitting up the MSTest reference here? Also this isn't ordered alphabetically anymore.

Comment (Measurity) [reply Nitrox.Test/Nitrox.Test.csproj:16]: Do we *manually* have to keep it alphabetic? All the changes I did were done through nuget package manager.

**Comment by Measurity** [comment]
I think I solved those issues now Tornac 🙏 .

Parallel builds will still show warnings on file locks, but it should work regardless. I will migrate the launcher over to .net 7 in linux branch which will make it better compatible with normal MSBuild flow (using project references).

**Pull request closed without merging**